### PR TITLE
fix: the item that removed is exist yet

### PR DESCRIPTION
**Pull request by FeiWang1119**
undock items that removed

Issus: https://github.com/linuxdeepin/developer-center/issues/4631

### CODE REVIEW

File: frame/taskmanager/dbushandler.cpp [hunk 1]
```diff
@@ -22,6 +22,13 @@ DBusHandler::DBusHandler(TaskManager *taskmanager, QObject *parent)
     , m_xEventMonitor(nullptr)
     , m_launcher(new org::deepin::dde::Launcher1(launcherService, launcherPath, QDBusConnection::sessionBus(), this))
 {
+    QDBusInterface *interAM = new QDBusInterface(ApplicationManager1DBusName, "/org/desktopspec/ApplicationManager1", "org.desktopspec.DBus.ObjectManager");
```
Comment (18202781743) [inline frame/taskmanager/dbushandler.cpp:25]: 内存泄漏，可以指定parent

File: frame/taskmanager/taskmanager.cpp [hunk 2]
```diff
@@ -91,6 +95,20 @@ TaskManager::TaskManager(QObject *parent)
...
+        QString desktopFile = DUtil::unescapeFromObjectPath(objectPath.path());
+        QString desktopName = desktopFile.split('/').last();
+        QList<Entry *> entries = m_entries->getEntries();
+        auto desktopEntryIter = std::find_if(entries.begin(), entries.end(), [desktopName] (Entry *entry) {
+             return entry->getDesktopFile().contains(desktopName);
+        });
+        if (desktopEntryIter != entries.end()) {
+            undockEntry(*desktopEntryIter);
+        } else {
+            qWarning() << "The entry which is to be removed is not found!";
```
Comment (18202781743) [inline frame/taskmanager/taskmanager.cpp:109]: 这应该是个debug信息吧，

File: frame/taskmanager/dbushandler.cpp (hunk 1 above)
Comment (FeiWang1119) [reply frame/taskmanager/dbushandler.cpp:25]: done

File: frame/taskmanager/taskmanager.cpp (hunk 2 above)
Comment (FeiWang1119) [reply frame/taskmanager/taskmanager.cpp:109]: 找不到就出问题了，图标就移除不了，算个警告。

**Review by 18202781743 — APPROVED** [review]

**Pull request merged**